src/PostgREST/Middleware.hs

```diff
@@ -54,26 +50,14 @@ runWithClaims conf app req = do
             else setRole anon : jwtEnv
    jwtEnv = claimsToSQL claims
 
-redirectInsecure :: Application -> Application
-redirectInsecure app req respond = do
-  let hdrs = requestHeaders req
-      host = lookup "host" hdrs
-      uriM = parseURI . cs =<< mconcat [
-        Just "https://",
-        host,
-        Just $ rawPathInfo req,
-        Just $ rawQueryString req]
-      isHerokuSecure = lookup "x-forwarded-proto" hdrs == Just "https"
-
+checkInsecure :: Application -> Application
+checkInsecure app req respond =
   if not (isSecure req || isHerokuSecure)
-    then case uriM of
-      Just uri ->
-        respond $ responseLBS status301 [
-            (hLocation, cs . show $ uri { uriScheme = "https:" })
-          ] ""
-      Nothing ->
-        respond $ responseLBS status400 [] "SSL is required"
+    then respond $ responseLBS status403 [] "SSL is required"
```

Comment:
Yeah `responseLBS` just passes a string along as the response, no matter what the status code. You'll need to hard code the json in there and probably set the content type response header to say it's json.